classes/mugoesitemplateoperators.php

```diff
@@ -42,8 +48,21 @@ function modify( $tpl, $operatorName, $operatorParameters, $rootNamespace, $curr
                 {
                     // Advertise this back to the reverse proxy
                     header( 'Surrogate-Control: abc=ESI/1.0' );
+                    ezjscPackerTemplateFunctions::setPersistentArray( 'surrogate', true, $tpl, false, false, false, true );

```

Comment:
Add a comment: "Advertise ESI to the pagelayout; otherwise, eZ Publish won't continue to pass the header from a full view template when view cache is turned on."